tts/preferences.py

```diff
@@ -3,40 +3,45 @@
 import tkinter.simpledialog as simpledialog
 import tkinter.filedialog as filedialog
 import tkinter.messagebox as messagebox
-import winreg
 import tts
+import platform
+import os
+
+if platform.system() == 'Windows':
+  import winreg
+else:
+  import xdgappdirs
+  import configparser
+
+class Preferences(object):
+  def __new__(cls):
```

Comment:
Externally clean, if internally a bit gunky way of doing it; I'd probably have gone with making Preferences abstract and added a static factory method to it to return the right sub-implementation.